matrix/lambdas/daemons/mapper.py

```diff
@@ -77,16 +78,21 @@ def _get_chunk_specs(bundle_uuid: str, bundle_version: str) -> typing.List[dict]
         :param bundle_version: Bundle version of the analysis bundle containing the expression matrix
         :return: List of dicts describing row subsets (chunks) of the input expression matrix
         """
-        zarr_store = DSSZarrStore(bundle_uuid,
-                                  bundle_version=bundle_version,
-                                  dss_instance=os.environ['DEPLOYMENT_STAGE'])
-        root = zarr.group(store=zarr_store)
-
-        rows_per_chunk = root.expression.chunks[0]
-        total_chunks = root.expression.nchunks
-
-        return [{"bundle_uuid": bundle_uuid,
-                 "bundle_version": bundle_version,
-                 "start_row": n * rows_per_chunk,
-                 "num_rows": rows_per_chunk}
-                for n in range(total_chunks)]
+
+        chunk_work_spec = []
+        for bundle_uuid, bundle_version in zip(bundle_uuids, bundle_versions):
+            zarr_store = DSSZarrStore(bundle_uuid,
+                                      bundle_version=bundle_version,
+                                      dss_instance=os.environ['DEPLOYMENT_STAGE'])
+            root = zarr.group(store=zarr_store)
+
+            rows_per_chunk = root.expression.chunks[0]
+            total_chunks = root.expression.nchunks
+
+            chunk_work_spec.extend(
+                [{"bundle_uuid": bundle_uuid,
+                  "bundle_version": bundle_version,
+                  "start_row": n * rows_per_chunk,
+                  "num_rows": rows_per_chunk}
+                 for n in range(total_chunks)])
```

Comment:
This still produces a flat array of worker specs, it just processes a list of bundles instead of a single bundle.